### PR TITLE
Fix test style in unused parentheses lint test

### LINKED ISSUE #63237: Fix test style in unused parentheses lint test
Update test file to use `#![deny(unused_parens)]` + `//~ ERROR` annotations instead of `pretty-json` (see https://rust-lang.github.io/rustc-guide/tests/adding.html)

The files need to update are:
- `test/lint/unused_parens_json_suggestions.rs`
- `test/lint/used_parens_remove_json_suggestion.rs`

This issue has been assigned to @polyedre via [this comment](https://github.com/rust-lang/rust/issues/63237#issuecomment-543196226).

### PR DESCRIPTION
**Pull request by polyedre**
I think this fixes #63237 
I'm not sure if I had to add text after the `//~ ERROR` comments.
This is my first pull request, so I'm open to feedback.
This issues already received one pull request [here](https://github.com/rust-lang/rust/pull/63257) but it was marked as closed for inactivity.

r?  @nikomatsakis

### CODE REVIEW

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
This looks great! I left one nit.

File: src/test/ui/lint/unused_parens_json_suggestion.rs [hunk 1]
```diff
@@ -8,13 +7,13 @@
...
 
-#![warn(unused_parens)]
+#![deny(unused_parens)]
 #![allow(unreachable_code)]
 
 fn main() {
     // We want to suggest the properly-balanced expression `1 / (2 + 3)`, not
     // the malformed `1 / (2 + 3`
-    let _a = (1 / (2 + 3));
+    let _a = (1 / (2 + 3)); //~ERROR
```
[the remark this answers is not in the document]
Comment (polyedre) [reply src/test/ui/lint/unused_parens_json_suggestion.rs]: It should be fixed by https://github.com/rust-lang/rust/pull/65507/commits/95d98af806f78d9cbe3975b633d5341394a155ad

**Comment by nikomatsakis** [comment]
@bors r+ rollup

Thanks @polyedre

**Review by nikomatsakis — APPROVED** [review]

**Pull request merged**